File: 2502-sort-the-people/sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        for (int i = 0; i < n - 1; i++) {
            int maxIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (heights[j] > heights[maxIndex]) {
                    maxIndex = j;
                }
            }
            String tempName = names[i];
            names[i] = names[maxIndex];
            names[maxIndex] = tempName;

            int tempHeight = heights[i];
            heights[i] = heights[maxIndex];
            heights[maxIndex] = tempHeight;
        }
        return names;
    }
}

File: 2502-sort-the-people/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        size_t n = names.size();
        
        for (size_t i{0}; i < n - 1; i++) 
        {
            int maxIndex = i;
            
            for (size_t j = i + 1; j < n; j++)
            {
                if (heights[j] > heights[maxIndex])
                {
                    maxIndex = j;
                }
            }
            
            swap(names[i], names[maxIndex]);
            swap(heights[i], heights[maxIndex]);
        }
        
        return names;
    }
};